### ISSUE TITLE
Classes from main source set are missing when running analysis for test source set.

### ISSUE THREAD

**Issue by smancill** [spotbugs/spotbugs-gradle-plugin#229]
When running `:spotbugsTest` the following output appears:

```
> Task :spotbugsTest
The following classes needed for analysis were missing:
<<list of all classes from main that are used in test>>
```

This is, all the classes from the `main` source set that are used by the tests in the `test` source set are reported as missing.

I have fixed it by adding the output from the `main` source set to `auxClassPaths` for the `spotbugsTest` task:

``` groovy
spotbugsTest {
    auxClassPaths += sourceSets.main.output
}
```

But I guess this should be set by the plugin itself. From a quick look to the code it looks that only the dependencies from `main` are added to the classpath but not the output of `main` itself.

Plugin version: 4.0.5\
SpotBugs version: 4.0.1\
Gradle version: 6.3

**Comment by KengoTODA** [comment]
could you share a [mcve](https://stackoverflow.com/help/minimal-reproducible-example) project, to reproduce it in my local?

**Comment by smancill** [comment]
Use Gradle itself to create a project skeleton:

    $ gradle init --dsl groovy --type java-library --test-framework junit

And then simply edit `build.gradle` to add the spotbugs plugin:

    id 'com.github.spotbugs' version '4.0.5'

And just build the project:

```
$ ./gradlew build
> Task :compileJava
> Task :processResources NO-SOURCE
> Task :classes
> Task :jar
> Task :assemble
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses
> Task :spotbugsMain

> Task :spotbugsTest
The following classes needed for analysis were missing:
  spotbugs.plugin.test.Library

> Task :test
> Task :check
> Task :build
```
 
The workaround that I put in the issue description solves the problem.

**Comment by KengoTODA** [comment]
I cannot reproduce the reported problem.

Confirmed that the `auxclasspath` option contains the `build/classes/java/main` which has `pkg/TargetClass.class`. And the generated report has `<Errors errors="0" missingClasses="0"></Errors>`.

```
2020-05-04T14:20:07.338+0800 [DEBUG] [com.github.spotbugs.snom.internal.SpotBugsRunner] Arguments for SpotBugs are generated: [
-sortByClass
-timestampNow
-auxclasspath
/private/tmp/a/build/classes/java/main:/private/tmp/a/build/resources/main:~/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-math3/3.6.1/e4ba98f1d4b3c80ec46392f25e094a6a2e58fcbf/commons-math3-3.6.1.jar:~/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/28.2-jre/8ec9ed76528425762174f0011ce8f74ad845b756/guava-28.2-jre.jar:~/.gradle/caches/modules-2/files-2.1/junit/junit/4.12/2973d150c0dc1fefe998f834810d68f278ea58ec/junit-4.12.jar:~/.gradle/caches/modules-2/files-2.1/com.google.guava/failureaccess/1.0.1/1dcf1de382a0bf95a3d8b0849546c88bac1292c9/failureaccess-1.0.1.jar:~/.gradle/caches/modules-2/files-2.1/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/b421526c5f297295adef1c886e5246c39d4ac629/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:~/.gradle/caches/modules-2/files-2.1/com.google.code.findbugs/jsr305/3.0.2/25ea2e8b0c338a877313bd4672d3fe056ea78f0d/jsr305-3.0.2.jar:~/.gradle/caches/modules-2/files-2.1/org.checkerframework/checker-qual/2.10.0/5786699a0cb71f9dc32e6cca1d665eef07a0882f/checker-qual-2.10.0.jar:~/.gradle/caches/modules-2/files-2.1/com.google.errorprone/error_prone_annotations/2.3.4/dac170e4594de319655ffb62f41cbd6dbb5e601e/error_prone_annotations-2.3.4.jar:~/.gradle/caches/modules-2/files-2.1/com.google.j2objc/j2objc-annotations/1.3/ba035118bc8bac37d7eff77700720999acd9986d/j2objc-annotations-1.3.jar:~/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar
-sourcepath
/private/tmp/a/src/test/resources:/private/tmp/a/src/test/java
-xml:withMessages
-outputFile
/private/tmp/a/build/reports/spotbugs/test.xml
-projectName
a (spotbugsTest)
-release
unspecified
-analyzeFromFile
/var/folders/f3/y4_b7t5j7q14_sfzfpdd9lb40000gn/T/spotbugs-gradle-plugin7920145707044290126.txt
]
```

**Comment by KengoTODA** [comment]
Ah, I found the root cause. `4.0.7` or older version has a problem around `check` task creation, and `4.0.8` fixed it so I cannot reproduce this issue in the version `4.0.8`.

Please use `4.0.8` then you can avoid this issue.
